### DISCUSSION TITLE
How do I know if the heartbeat connection is broken?

### DISCUSSION THREAD

**Discussion by accountnujen** [curl/curl#13138] [Q&A]
I make a request to my camera, which responds every 5 seconds with new events. 
curl --digest --user admin:admin -g --request GET "192.168.0.108/cgi-bin/eventManager.cgi?action=attach&codes=[All]&heartbeat=5".
If there are no events, what I see is this:
```
--myboundary
Content-Type: text/plain
Content-Length: 9
Heartbeat

--myboundary
Content-Type: text/plain
Content-Length: 9
Heartbeat

--myboundary
Content-Type: text/plain
Content-Length: 9
Heartbeat
```

I need to maintain a perpetual connection, however if the connection is broken (camera disconnected), I won't know about it because no errors will occur. Moreover, if the camera is connected - my request will not continue. How do I know if the connection has been broken?

**Comment by dfandrich** [comment]
Setting CURLOPT_LOW_SPEED_LIMIT and CURLOPT_LOW_SPEED_TIME are probably your
best bets, or CURLOPT_TCP_KEEPALIVE.
